app/src/main/java/com/example/android/oscarine/SharedPreference.kt

```diff
@@ -0,0 +1,25 @@
+package com.example.android.oscarine
+
+import android.content.Context
+import android.content.SharedPreferences
+import android.util.Log
+
+class SharedPreference(context: Context) {
+
+    private val preferenceName: String = context.getResources().getString(R.string.SharedPreferencesFilename)
```

Comment:
You should only store translatable and dimension dynamic strings in strings.xml, not constants

Reply:
Will this work then?
```suggestion
    private val preferenceName = "com.example.android.oscarine.prefs"
```

Reply:
Yes, but something simpler, like `oscarine_preferences`